### DISCUSSION TITLE
Proposed feature: network jacks/port

### DISCUSSION THREAD

**Discussion by wannagofast83** [netbox-community/netbox#17739] [Ideas]
I was thinking about submitting a feature request and wanted to get others thoughts on it.

Proposed functionality:
It would be nice Similar to front/rear port, it would be nice if there was a way to track jacks in the field associated to a site. For example, if site X has jack 1234 in room 400, we would document that accordingly and connect it to an interface/front/rear port.

Use case:
This would show user connections in the field and provide documentation for the rooms and interface that the port is connected to.

I feel that this is similar to front/rear ports but I don't think that the jacks in the field  would require rear port for the cable run.

Has anyone else had a need for something like this?

**Comment by MattDRogers** [comment]
If I were in an IT department and needed to track PCs, printers, VoIP phones and such to the wall-mounted jacks in a room/suite back to the data closet where they patch through to a switch, then yes this would be a very useful feature to request.

It can be done today, but obviously it's not gonna be a 1:1 thing. You'd have to translate reality into the current limitations of the data model. Where in the real world you have a room with wall-mounted jacks, in NetBox you'd have a Rack with a Device for each wall-mounted jack and Devices that represent computers, printers, VoIP phones, etc.

You could still do the front and rear port thing if you want to trace cabling from say a computer to a switch in a data closet. A wall jack is basically a patch-panel, after all. A cable terminates to the rear side, and another cable connects to the front side.

I'm not sure what the new data model would look like, other than to say that the "room" would be on the same level as Racks occupy currently. Maybe? The Location model is the closest to "rooms" that we have now (at least the way I use it). But, you can't put a Device in a Location. Device only go into Racks. Hmm...

I think this may be out-of-scope for NetBox (unless doing the "Rack as a room approach"), but it is a cool idea. Fun to think about.

Also, I'm sorry if I missed the point. But you got me thinking :)

**Reply by wannagofast83** [reply]
I agree. I was thinking that the jack would be "loose" in the site. I have access points that are in Netbox just in the site, not a location or rack. I created custom fields for some of the details that Netbox doesn't track such as room number. 

The front rear ports would work, just troublesome to import the rear then import the front. I wish there was an easier way for both of those.

**Reply by xkilian** [reply]
A location represents anything you want: room, building, floor. You can place devices in locations directly.
What uou want already exists. It just lacks tje x/y/z 3d placement, but this can be mitigated with custom fields.
Hope this helps.

**Reply by wannagofast83** [reply]
That's valid. We have over 300 buildings and I don't think that it would be pratcial or helpful to create a location for each room number. I like the idea of doing it per floor though.
